app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,55 @@
+class SessionsController < ApplicationController
+
+
+  get '/' do
+    if logged_in?
+      @user = User.find_by_id(session[:id])
+    end
+    erb :home
+  end
+
+
+  get '/signup' do
+    if logged_in?
+      redirect '/tweets'
+    else
+      erb :'registrations/signup'
+    end
+  end
+
+  get '/login' do
+    if logged_in?
+      redirect '/tweets'
+    else
+      erb :'sessions/login'
+    end
+  end
+
+  get '/logout' do
+    if logged_in?
+      logout!
+      redirect '/login'
+    else
+      redirect '/'
+    end
+  end
+
+  post '/login' do
+    binding.pry
+    @user = User.find_by(username: params[:username])
+      session[:id] = @user.id
+      erb 'tweets/tweets'
+  end
+
+  post '/signup' do
+    if params[:username].empty? || params[:email].empty? || params[:password].empty?
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.